app/models/ebean/VersionSpecification.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2016 Groupon.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package models.ebean;
+
+import com.avaje.ebean.annotation.CreatedTimestamp;
+import com.avaje.ebean.annotation.UpdatedTimestamp;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import java.sql.Timestamp;
+
+/**
+ * Data model for Version Specifications (for Evergreen).
+ *
+ * @author Matthew Hayter (mhayter at groupon dot com)
+ */
+@Entity
+@Table(name = "version_specifications", schema = "portal")
+public class VersionSpecification {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id")
+    private Long id;
+
+    @CreatedTimestamp
+    @Column(name = "created_at")
+    private Timestamp createdAt;
+
+    @UpdatedTimestamp
+    @Column(name = "updated_at")
+    private Timestamp updatedAt;
+
+    /*
+    Note that these precedence values are intended to be sequential; each new row inserted with some precedence will force the update
```

Comment:
Maybe I'm not seeing it, but this method seems to be more complex. When someone does a rule insertion you will need to change N - K entries in the database where K is the index to which the rule was added. The linked list solution seems the most straight forward where each row in the database has a link to the next rule and the last one has null. In either case the internal model should just be an ordered list, it's just a question of what the storage model will be. 

From a user perspective, will each edit be applied at once (e.g. add row here, change row content, delete row, etc.) or will it be change all the rows and save? Of course we can do either on the front-end (and should retain the possibility to change) and map it correctly to the back-end. Although, in the save everything case we need to overwrite the entire table anyway unless we're going to diff to figure out the increments.

So I don't think I have a problem with the write everything philosophy unless @BrandonArp has any objections to it.

Finally, I'm sure you will implement some sort of optimistic locking to make this nice and sane ;)

Ref:
http://stackoverflow.com/questions/21388/represent-ordering-in-a-relational-database